File: solutions/csharp/collatz-conjecture/2/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number < 1) throw new ArgumentOutOfRangeException("I can't do that Dave.");
        var counter = 0;
        while (number > 1)
        {
            counter++;
            if (number % 2 == 0) number /= 2;
            else number = (number * 3) + 1;
        }
        return counter;
    }
}

File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
    if (start < 1) return -1;
    int counter = 0;
    while (start > 1)
    {
        counter++;
        if (start % 2 == 0) start /= 2;
        else start = (start * 3) + 1;
    }
    return counter;
}